frontend/src/components/FilePreview/index.tsx

```diff
@@ -0,0 +1,127 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { FilePreview as AutoFilePreview } from '@autonomys/auto-design-system';
+import {
+  OffchainMetadata,
+  canDisplayDirectly,
+  needsContentParsing,
+  processFileData,
+  decryptFileData,
+} from '@autonomys/auto-dag-data';
+import { useNetwork } from '../../contexts/network';
+import { NetworkId as AutoDriveNetworkId } from '../../constants/networks';
+import { NetworkId as AutoUtilsNetworkId } from '@autonomys/auto-utils';
+import { EXTERNAL_ROUTES } from '../../constants/routes';
+
+export const FilePreview = ({ metadata }: { metadata: OffchainMetadata }) => {
+  const { network } = useNetwork();
+  const [isFilePreview, setIsFilePreview] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+  const [isDecrypted, setIsDecrypted] = useState(false);
+  const [decryptionError, setDecryptionError] = useState<string | null>(null);
+  const [file, setFile] = useState<Blob | null>(null);
+  const [textContent, setTextContent] = useState<string | null>(null);
+
+  const networkId = useMemo(() => {
+    switch (network.id) {
+      case AutoDriveNetworkId.LOCAL:
+      case AutoDriveNetworkId.MAINNET:
+        return AutoUtilsNetworkId.MAINNET;
+      case AutoDriveNetworkId.TAURUS:
+        return AutoUtilsNetworkId.TAURUS;
+      default:
+        return AutoUtilsNetworkId.MAINNET;
+    }
+  }, [network]);
+
+  const gatewayUrl = EXTERNAL_ROUTES.gatewayObjectDownload(metadata.dataCid);
```

Comment:
Should we validate that this is a valid CID prior to making the request?

Reply:
I don't think so, the cid passed (`metadata.dataCid`) to the method comes from the metadata structure